crates/sound/src/audio_backends/alsa.rs

```diff
@@ -145,17 +145,18 @@ fn write_samples_direct(
         let Some(buffer) = stream.buffers.front_mut() else {
             return Ok(false);
         };
-        let mut buf = vec![0; buffer.data_descriptor.len() as usize];
-        let read_bytes = buffer
-            .consume(&mut buf)
-            .expect("failed to read buffer from guest");
+        let mut buf = vec![0; buffer.len];
+        let read_bytes = buffer.consume(&mut buf).unwrap_or_else(|_| {
+            log::error!("Failed to read from guest memory");
+            0
```

Comment:
This .unwrap + log but continue seems to be a common pattern here? Is there a reason why we can't just error out of the current function (and then, maybe log it at a higher level from some central place)?